bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/DecryptionMediator.cs

```diff
@@ -219,24 +198,96 @@ List<CiphertextBallot> ballots
         }
     }
 
-    protected override void DisposeUnmanaged()
+    #endregion
+
+    #region Accumulate Tally
+
+    // accumulate can be called before CreateChallenge
+    // but it doesnt have to be because CreateChallenge will also handle it
+    public void AccumulateShares(string tallyId, bool skipValidation = false)
+    {
+        _ = TallyDecryptions[tallyId].AccumulateShares(skipValidation);
+    }
+
+    #endregion
+
+    #region Challenge
+
+    public Dictionary<string, GuardianChallenge> CreateChallenge(string tallyId, bool skipValidation = false)
+    {
+        return TallyDecryptions[tallyId].CreateChallenge(skipValidation);
+    }
+
+    #endregion
+
+    #region Challenge Response
+
+    public void SubmitResponse(
+        string tallyId, GuardianChallengeResponse response)
+    {
+        TallyDecryptions[tallyId].SubmitChallengeResponse(response);
+    }
+
+    public bool ValidateResponses(string tallyId)
+    {
+        return TallyDecryptions[tallyId].ValidateChallengeResponses();
+    }
+
+    // compute proofs can be called before decrypt
+    // but it doesnt have to be because decrypt will also handle it
+    public void ComputeProofs(string tallyId, bool skipValidation = false)
```

Comment:
similarly you dont have to call compute proofs, but you can if performance necessitates